xellgrid/grid.py

```diff
@@ -953,6 +957,27 @@ def _add_empty_row(self, row_index):
         update_table(self, triggered_by='add_empty_row',
                            scroll_to_row=self._df.index.get_loc(row_index))
 
+    def _replace_df_by_data(self, data):
```

Comment:
Will this cause unexpected issue? Once a user uploads a local file, it will automatically replace the existing df in Grid? It may be unexpected from user perspective.